src/handlers/GraphQLHandler.test.ts

```diff
@@ -82,132 +82,49 @@ describe('info', () => {
     expect(handler.info).toHaveProperty('operationName', 'Login')
   })
 
-  test('parses operation name out of DocumentNode for query', () => {
-    const handler = new GraphQLHandler(
-      'query',
-      GetUserDetailDocument,
-      '*',
-      resolver,
-    )
+  test('parses a query operation name from a given DocumentNode', () => {
+    const node = parse(`
+      query GetUser {
+        user {
+          firstName
+        }
+      }
+    `)
 
-    expect(handler.info).toHaveProperty(
-      'header',
-      'query GetUserDetail (origin: *)',
-    )
+    const handler = new GraphQLHandler('query', node, '*', resolver)
+
+    expect(handler.info).toHaveProperty('header', 'query GetUser (origin: *)')
     expect(handler.info).toHaveProperty('operationType', 'query')
-    expect(handler.info).toHaveProperty('operationName', 'GetUserDetail')
+    expect(handler.info).toHaveProperty('operationName', 'GetUser')
   })
 
-  test('parses operation name out of DocumentNode for mutation', () => {
-    const handler = new GraphQLHandler('mutation', LoginDocument, '*', resolver)
+  test('parses a mutation operation name from a given DocumentNode', () => {
+    const node = parse(`
```

Comment:
I'm replacing the manual `DocumentNode` instances with the invocations of the `parse` function from GraphQL. This way we retain the test scenario while keeping it much more concise.